src/OpenLoco/src/Objects/StreetLightObject.cpp

```diff
@@ -24,11 +28,21 @@ namespace OpenLoco
     // 0x00477F19
     void StreetLightObject::load(const LoadedObjectHandle& handle, stdx::span<const std::byte> data, ObjectManager::DependentObjects*)
     {
-        Interop::registers regs;
-        regs.esi = Interop::X86Pointer(this);
-        regs.ebx = handle.id;
-        regs.ecx = enumValue(handle.type);
-        Interop::call(0x00477F19, regs);
+        auto remainingData = data.subspan(sizeof(StreetLightObject));
+
+        // Load object name string
+        auto strRes = ObjectManager::loadStringTable(remainingData, handle, 0);
+        name = strRes.str;
+        remainingData = remainingData.subspan(strRes.tableLength);
+
+        // TODO: verify what's happening at 0x00477F31
```

Comment:
nothing. I suspect originally streetlights had a dependent object but then it was changed at a later date. TODO can be removed.

Reply:
Alright, thanks for checking!